[CI] RemoteClusterServiceTests.testCollectNodes fails

Fails on 7.x:
https://gradle-enterprise.elastic.co/s/fggxy7emgdb7a

Cannot reproduce locally with
```
./gradlew ':server:test' --tests "org.elasticsearch.transport.RemoteClusterServiceTests.testCollectNodes" \
  -Dtests.seed=E8A58760A64ADCFF \
  -Dtests.security.manager=true \
  -Dtests.locale=en-CA \
  -Dtests.timezone=Canada/Pacific \
  -Dcompiler.java=14
```

Error in Logs:
```
java.lang.AssertionError: 
Expected: “no seed node left”
but: was “Unable to open any connections to remote cluster [cluster_2]”
at __randomizedtesting.SeedInfo.seed([E8A58760A64ADCFF:D2512FB984274F6]:0)
at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
at org.junit.Assert.assertThat(Assert.java:956)
at org.junit.Assert.assertThat(Assert.java:923)
at org.elasticsearch.transport.RemoteClusterServiceTests.testCollectNodes(RemoteClusterServiceTests.java:661)
```

--- a/server/src/test/java/org/elasticsearch/transport/RemoteClusterServiceTests.java
+++ b/server/src/test/java/org/elasticsearch/transport/RemoteClusterServiceTests.java
@@ -651,8 +651,9 @@ public class RemoteClusterServiceTests extends ESTestCase {
                             });
                         failLatch.await();
                         assertNotNull(ex.get());
-                        if (ex.get() instanceof  IllegalStateException) {
-                            assertThat(ex.get().getMessage(), equalTo("no seed node left"));
+                        if (ex.get() instanceof IllegalStateException) {
+                            assertThat(ex.get().getMessage(), either(equalTo("no seed node left"))
+                                .or(equalTo("Unable to open any connections to remote cluster [cluster_2]")));
                         } else {
                             assertThat(ex.get(),
                                 either(instanceOf(TransportException.class)).or(instanceOf(NoSuchRemoteClusterException.class)));

